FetchRequests/Sources/Controller/CollapsibleSectionsFetchedResultsController.swift

```diff
@@ -418,33 +418,73 @@ private class DelegateThunk<FetchedObject: FetchableObject> {
 
     private weak var parent: (any Parent)?
 
+#if compiler(<6)
+    private let willChange: @MainActor (_ controller: Controller) -> Void
```

Comment:
Marking these as @sendable in Swift 6 is awkward since it would require `Parent` to be sendable.
That's why I can't use the old Thunk impl in Swift 6